email_alert_service/models/unpublishing_alert.rb

```diff
@@ -0,0 +1,35 @@
+require "gds_api/email_alert_api"
+
+class UnpublishingAlert
+  def initialize(document, logger, unpublishing_scenario)
+    @document = document
+    @logger = logger
+    @unpublishing_scenario = unpublishing_scenario
+  end
+
+  def trigger
+    logger.info "Received unsubscription notification for #{document['title']}, unpublishing_scenario: #{unpublishing_scenario}, full payload: #{document}"
+    lock_handler.with_lock_unless_done do
+      Services.email_api_client.bulk_unsubscribe(
+        {
+          content_id: document.fetch("content_id"),
+          body: unpublishing_scenario,
+        }.to_json,
```

Comment:
This takes the subscriber list ID, so we'll have to query email-alert-api to find that first.

Reply:
Yep, not touched this since you wrote your api-endpoint.

I'll get the api-adaptors merged then revisit this